shared/lib_time.cpp

```diff
@@ -83,14 +83,14 @@ void single_year_to_lifetime_interpolated(
 		// Scale single year interpolated vector to lifetime
 		for (size_t y = 0; y < n_years; y++) {
 			for (size_t i = 0; i < n_rec_single_year; i++) {
-				lifetime_from_singleyear_vector.push_back(singleyear_sampled[i] * scale_factor[y]);
+				lifetime_from_singleyear_vector.push_back(singleyear_sampled[i] * (scale_factor.size() > y ?  scale_factor[y] : 1.0));
```

Comment:
I think this work-around of a too-small scale_factor vector is a step in the right direction of flexibility, but I'd prefer more explicit handling of what's expected from this new scale_factor parameter as this seems to potentially hide errors.

@brtietz What are expected lengths of scale_factor? Same length of `singleyear_vector`, length 0 and length 1 seem reasonable to me.

Reply:
The original intent was `n_years`, which seems to have caused problems with PVWatts. scale_factors translates from the SSC inputs (which can be length 1 or length `n_years`). I'd prefer to avoid duplicating the code that's already in common.cpp, if possible.